### PR TITLE
complain if specified config files are missing

### PR DESCRIPTION
**Pull request by boegel**
fix for #1239

@stdweird: please review

### CODE REVIEW

File: easybuild/tools/options.py [hunk 1]
```diff
@@ -79,9 +80,10 @@ class EasyBuildOptions(GeneralOption):
     VERSION = this_is_easybuild()
 
     DEFAULT_LOGLEVEL = 'INFO'
-    DEFAULT_CONFIGFILES = DEFAULT_SYS_CFGFILES + [DEFAULT_USER_CFGFILE]
+    DEFAULT_CONFIGFILES = [p for p in DEFAULT_SYS_CFGFILES + [DEFAULT_USER_CFGFILE] if os.path.exists(p)]
```
Comment (stdweird) [inline easybuild/tools/options.py]: only test the `DEFAULT_USER_CFGFILE` (the other comes from a `glob`)

**Comment by stdweird** [comment]
@boegel the main difference is that `--help` won't show the locations of the default config files anymore. so maybe that needs an option (is there a `--show-defaults` btw?)?

**Comment by boegel** [comment]
@stdweird: that was a problem already, since `DEFAULT_SYS_CFGFILES` uses a `glob`...

There's no `--show-defaults` no, that's the job of `GeneralOption`, no?

**Comment by boegel** [comment]
@stdweird: the default paths being considered for configuration files are clearly documented at http://easybuild.readthedocs.org/en/latest/Configuration.html#default-configuration-files, I doubt a lot of users rely on what `--help` lists as a 'default'

**Comment by stdweird** [comment]
`--show-defaults` here means: show all paths EB considers. go already shows the defautl sin the help. but in this case EB, preprocesses the defaults (and that's probably the reason why the default was set to allow missing config files). config files that are specirfied on the commandline / environment shouldn't be missing tough

**Comment by boegel** [comment]
Do you mean specifically for `--configfiles`, or across all path-related options?

**Comment by stdweird** [comment]
config files, there is no such thing as `path-related options` in generaloption

**Comment by stdweird** [comment]
@boegel i gave it some more thought. i'm going to implement what i described above: an option to only fail on missing config files that have been set explicitly via command or environment (and not fail on the default ones; unless the default ones are specified on commandline/environment)
then you can still choose which one to enforce.

but generaloption should fail when someone does `--configfiles=/wrong/path` by default imho. there was no existing code path to do that, but i think the required bit are available since `store_or_None`, so not that much work

**Comment by boegel** [comment]
@stdweird: makes sense, that would make things significantly simpler here

While you're at it, how about adding support for specifying config files via a glob pattern (that may contain environment variables)?
Then I can make the default paths in EasyBuild truly static, by using (the equivalent of) `$XDG_CONFIG_DIR/easybuild.d/*.cfg` and letting generaloption figuring it out.

Maybe I'm asking too much here though, especially the env var part. ;-)

But, to make the default list of configuration files truly static (and make the default mentioned in `--help` output clear), that's what I would need...

**Comment by boegel** [comment]
Added `--show-default-configfiles` as requested by @stdweird, example output:

```
$ XDG_CONFIG_DIRS=/tmp/etc:/tmp/moaretc eb --show-default-configfiles
Couldn't import dot_parser, loading of dot files will not be possible.
Default list of configuration files:

[with $XDG_CONFIG_HOME: (not set), $XDG_CONFIG_DIRS: /tmp/etc:/tmp/moaretc]

* user-level: ${XDG_CONFIG_HOME:-$HOME/.config}/easybuild/config.cfg
  -> /Users/kehoste/.config/easybuild/config.cfg => found
* system-level: ${XDG_CONFIG_DIRS:-/etc}/easybuild.d/*.cfg
  -> {/tmp/etc, /tmp/moaretc}/easybuild.d/*.cfg => /tmp/etc/easybuild.d/config.cfg, /tmp/moaretc/easybuild.d/bar.cfg, /tmp/moaretc/easybuild.d/foo.cfg

Default list of existing configuration files (4): /tmp/etc/easybuild.d/config.cfg, /tmp/moaretc/easybuild.d/bar.cfg, /tmp/moaretc/easybuild.d/foo.cfg, /Users/kehoste/.config/easybuild/config.cfg
```

**Comment by boegel** [comment]
I'm not going to hold this back any longer. @stdweird: if you end up implementing the support for only complaining if specified config files are non-existent, we can update the code here accordingly

The only 'issue' now is that the default value for `--configfiles` as shown by `eb --help` as preprocessed in the sense that only *existing* config files are chosen; this is remedied by the documentation available at http://easybuild.readthedocs.org/en/latest/Configuration.html#default-configuration-files, and now also with `--show-default-configfiles`. So I don't consider that a blocker (especially since this PR doesn't change this, it was already like that).

**Pull request merged**